examples/SplitKernels/gammas.cu

```diff
@@ -110,122 +113,149 @@ __global__ void TransportGammas(Track *gammas, const adept::MParray *active, Sec
+      electron.dir      = energy * currentTrack.dir - newEnergyGamma * newDirGamma;
+      electron.dir.Normalize();
+    } else {
+      atomicAdd(&globalScoring->energyDeposit, energyEl);
+      atomicAdd(&scoringPerVolume->energyDeposit[volumeID], energyEl);
+    }
 
-      positron.InitAsSecondary(/*parent=*/currentTrack);
-      // Reuse the RNG state of the dying track.
-      positron.rngState = currentTrack.rngState;
-      positron.energy   = posKinEnergy;
-      positron.dir.Set(dirSecondaryPos[0], dirSecondaryPos[1], dirSecondaryPos[2]);
+    // Check the new gamma energy and deposit if below threshold.
+    if (newEnergyGamma > LowEnergyThreshold) {
+      currentTrack.energy = newEnergyGamma;
+      currentTrack.dir    = newDirGamma;
 
+      // The current track continues to live.
+      activeQueue->push_back(globalSlot);
+    } else {
+      atomicAdd(&globalScoring->energyDeposit, newEnergyGamma);
+      atomicAdd(&scoringPerVolume->energyDeposit[volumeID], newEnergyGamma);
       // The current track is killed by not enqueuing into the next activeQueue.
-      break;
     }
-    case 1: {
-      // Invoke Compton scattering of gamma.
-      constexpr double LowEnergyThreshold = 100 * copcore::units::eV;
-      if (energy < LowEnergyThreshold) {
-        activeQueue->push_back(slot);
-        continue;
-      }
-      const double origDirPrimary[] = {currentTrack.dir.x(), currentTrack.dir.y(), currentTrack.dir.z()};
-      double dirPrimary[3];
-      const double newEnergyGamma =
-          G4HepEmGammaInteractionCompton::SamplePhotonEnergyAndDirection(energy, dirPrimary, origDirPrimary, &rnge);
-      vecgeom::Vector3D<double> newDirGamma(dirPrimary[0], dirPrimary[1], dirPrimary[2]);
-
-      const double energyEl = energy - newEnergyGamma;
-      if (energyEl > LowEnergyThreshold) {
-        // Create a secondary electron and sample/compute directions.
-        Track &electron = secondaries.electrons.NextTrack();
-        atomicAdd(&globalScoring->numElectrons, 1);
-
-        electron.InitAsSecondary(/*parent=*/currentTrack);
-        electron.rngState = newRNG;
-        electron.energy   = energyEl;
-        electron.dir      = energy * currentTrack.dir - newEnergyGamma * newDirGamma;
-        electron.dir.Normalize();
-      } else {
-        atomicAdd(&globalScoring->energyDeposit, energyEl);
-        atomicAdd(&scoringPerVolume->energyDeposit[volumeID], energyEl);
-      }
+  } else if constexpr (ProcessIndex == 2) {
+    // Invoke photoelectric process.
+    const double theLowEnergyThreshold = 1 * copcore::units::eV;
 
-      // Check the new gamma energy and deposit if below threshold.
-      if (newEnergyGamma > LowEnergyThreshold) {
-        currentTrack.energy = newEnergyGamma;
-        currentTrack.dir    = newDirGamma;
+    const double bindingEnergy = G4HepEmGammaInteractionPhotoelectric::SelectElementBindingEnergy(
+        &g4HepEmData, theMCIndex, soaData.gamma_PEmxSec[soaSlot], energy, &rnge);
```

Comment:
`soaData.gamma_PEmxSec` is never set, so very likely this is reading uninitialized memory...

Reply:
Ah, thanks! I was already aware that there was a mistake when applying the patch, but I didn't have the time yesterday to review the diff! You saved me some work here!